src/helpers/presentation.rs

```diff
@@ -0,0 +1,323 @@
+use regex::Regex;
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+
+use thiserror::Error;
+use vade_evan_bbs::{CredentialSchema, RequestProofPayload};
+
+use super::{
+    datatypes::DidDocumentResult,
+    shared::{convert_to_nquads, create_draft_credential_from_schema, SharedError},
+};
+use crate::api::VadeEvan;
+use crate::helpers::datatypes::EVAN_METHOD;
+
+#[derive(Error, Debug)]
+pub enum PresentationError {
+    #[error("internal VadeEvan call failed; {0}")]
+    VadeEvanError(String),
+    #[error("{0}")]
+    JsonLdHandling(#[from] SharedError), // for now only one type of error from shared helper, so just re-package it
+    #[error("could not find revealed attributes in nquads: {0}")]
+    InvalidRevealedAttributes(String),
+    #[error("internal error occurred; {0}")]
+    InternalError(String),
+    #[error(r#"JSON serialization of {0} failed due to "{1}""#)]
+    JsonSerialization(String, String),
+    #[error(r#"JSON deserialization of {0} failed due to "{1}" on: {2}"#)]
+    JsonDeserialization(String, String, String),
+    #[error(r#"schema with DID "{0}" does not seem to be a valid schema: {1}"#)]
+    SchemaInvalid(String, String),
+    #[error(r#"schema with DID "{0}" could not be found"#)]
+    SchemaNotFound(String),
+}
+
+impl PresentationError {
+    fn to_serialization_error(
+        parsed_object: &str,
+    ) -> impl Fn(serde_json::Error) -> PresentationError + '_ {
+        move |err| PresentationError::JsonSerialization(parsed_object.to_string(), err.to_string())
+    }
+
+    fn to_deserialization_error<'a>(
+        to_parse_name: &'a str,
+        to_parse_value: &'a str,
+    ) -> impl Fn(serde_json::Error) -> PresentationError + 'a {
+        move |err| {
+            PresentationError::JsonDeserialization(
+                to_parse_name.to_string(),
+                err.to_string(),
+                to_parse_value.to_string(),
+            )
+        }
+    }
+}
+
+// Master secret is always incorporated, without being mentioned in the credential schema
+const ADDITIONAL_HIDDEN_MESSAGES_COUNT: usize = 1;
+const NQUAD_REGEX: &str = r"^_:c14n0 <http://schema.org/([^>]+?)>";
+const TYPE_OPTIONS: &str = r#"{ "type": "bbs" }"#;
+
+pub struct Presentation<'a> {
+    vade_evan: &'a mut VadeEvan,
+}
+
+impl<'a> Presentation<'a> {
+    pub fn new(vade_evan: &'a mut VadeEvan) -> Result<Self, PresentationError> {
+        Ok(Self { vade_evan })
+    }
+
+    pub async fn create_proof_request(
```

Comment:
Do we want to add function headers to all functions? Specifically for this one a bit duplicated because its already documented in the api files. Just asking :)

Reply:
Hm, we could. Was comparing with similar credential helper functions, and those didn't have documentation. But it seems like 3 of the 10 are documented. Will add documentation here. ^^